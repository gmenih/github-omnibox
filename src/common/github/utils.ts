export function toQueryString (obj: object): string {
    return Object.entries(obj)
        .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
        .join('&');
}
