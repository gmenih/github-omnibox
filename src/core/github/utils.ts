import {Primitive} from '../types/primitive';

export function toQueryString(obj: Record<string, Primitive>): string {
    return Object.entries(obj)
        .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
        .join('&');
}
