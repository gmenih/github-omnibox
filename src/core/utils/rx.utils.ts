import {Observable} from 'rxjs';

export function fromBrowserEvent<TCallback extends (...args: any[]) => void>(
    event: chrome.events.Event<TCallback>,
): Observable<Parameters<TCallback>> {
    return new Observable<Parameters<TCallback>>((subscriber) => {
        const eventCallback = (...args: Parameters<TCallback>): void => {
            subscriber.next(args);
        };

        event.addListener(eventCallback as TCallback);
        subscriber.add(() => event.removeListener(eventCallback as TCallback));
    });
}
