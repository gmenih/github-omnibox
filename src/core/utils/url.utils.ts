export function toQueryString(obj: Record<string, string>): string {
    return new URLSearchParams(obj).toString();
}
