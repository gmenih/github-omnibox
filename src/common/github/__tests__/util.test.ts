import {toQueryString} from '../utils';

describe('Github Util', () => {
    test.each([
        {this: 'object', to: 'query'},
        {should: 'properly', convert: 15, things: true},
        {even: 'weird\nvalues', like: true},
    ])('should properly convert to queryString', (object: any) => {
        expect(toQueryString(object)).toMatchSnapshot();
    });
});
