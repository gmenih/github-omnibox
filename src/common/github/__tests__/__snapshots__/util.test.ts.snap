// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github Util should properly convert to queryString 1`] = `"this=object&to=query"`;

exports[`Github Util should properly convert to queryString 2`] = `"should=properly&convert=15&things=true"`;

exports[`Github Util should properly convert to queryString 3`] = `"even=weird%0Avalues&like=true"`;
